test/pages/box_buster.spec.js

```diff
@@ -0,0 +1,290 @@
+import BootstrapVue from 'bootstrap-vue'
+import { mount, createLocalVue } from '@vue/test-utils'
+import flushPromises from 'flush-promises'
+import BoxBuster from '@/pages/box_buster.vue'
+import { getPlatesFromBoxBarcodes } from '@/modules/labwhere'
+import lighthouse from '@/modules/lighthouse_service'
+
+// Mock the whole module. Returning jest.fn() allows you to mock methods here
+// jest.mock('@/modules/labwhere', () => jest.fn())
+jest.mock('@/modules/labwhere')
+jest.mock('@/modules/lighthouse_service')
+
+// When rendering HTML browsers compress whitespace
+// When writing tests, we are more concerned with what the users sees, and
+// don't really care about additional whitespace introduced due to the
+// layout of our templates. This compresses all whitespace down.
+const squish = (text) => text.replace(/\s+/g, ' ')
+
+const localVue = createLocalVue()
+
+// List of example plates to use in testing.
+const plateA = {
+  plate_barcode: 'AP-rna-1-8-posi',
+  plate_map: true,
+  number_of_positives: 8
+}
+const plateB = {
+  plate_barcode: 'AP-rna-2-2-posi',
+  plate_map: true,
+  number_of_positives: 2
+}
+const plateC = {
+  plate_barcode: 'AP-rna-3-0-nmap',
+  plate_map: false,
+  number_of_positives: null
+}
+const plateD = {
+  plate_barcode: 'AP-rna-4-5-posi',
+  plate_map: true,
+  number_of_positives: 5
+}
+const plateE = {
+  plate_barcode: 'AP-rna-5-0-posi',
+  plate_map: true,
+  number_of_positives: 0
+}
+const examplePlates = [plateA, plateB, plateC, plateD, plateE]
+const expectedPlateTotal = 5
+const expectedMapTotal = 4
+const expectedMaplessTotal = 1
+const expectedPositiveTotal = 15
+
+localVue.use(BootstrapVue)
+
+describe('BoxBuster', () => {
+  let wrapper
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('is a Vue instance', () => {
+    const wrapper = mount(BoxBuster, { localVue })
+    expect(wrapper.findComponent(BoxBuster).exists()).toBeTruthy()
+  })
+
+  it('renders a list of plates', async () => {
+    const data = { plates: examplePlates }
+    const wrapper = mount(BoxBuster, { localVue })
+    await wrapper.setData(data)
+    const tableText = wrapper.find('table').text()
+    expect(tableText).toContain('AP-rna-1-8-posi')
+    expect(tableText).toContain('AP-rna-4-5-posi')
+    expect(tableText).toContain('AP-rna-2-2-posi')
+    expect(tableText).toContain('AP-rna-3-0-nmap')
+  })
+
+  it('shows a table with the expected headers', async () => {
+    const wrapper = mount(BoxBuster, { localVue })
+    await wrapper.setData({ labwhereResponse: { success: true } })
+    const header = wrapper.find('table').findAll('th')
+    expect(header.at(0).text()).toContain('Barcode')
+    expect(header.at(1).text()).toContain('Plate Map')
+    expect(header.at(2).text()).toContain('Number Of Positives')
+  })
+
+  it("only shows the table once we've scanned the Box", async () => {
+    const wrapper = mount(BoxBuster, { localVue })
```

Comment:
From what I remember setdata will be deprecated in the future so you would add the data when you mount but was wondering why you are having to set the data afterwards and do an await? From what I see setting the data is not async? Maybe I am missing something.

Reply:
I think you're thinking of set props, setData isn't flagged as deprecated:
https://vue-test-utils.vuejs.org/api/wrapper/setdata.html
You'll also notice that it uses await in the example.

It used to be that vue-test-utils would try to use magic to avoid the need to await, but they removed that ability as it was very flaky. The await in this case isn't so much the data update I believe, but for the subsequent render update.